selfdrive/car/tesla/carcontroller.py

```diff
@@ -40,23 +41,24 @@ def update(self, CC, CS, now_nanos):
 
     # Longitudinal control (in sync with stock message, about 40Hz)
     if self.CP.openpilotLongitudinalControl:
+      acc_state = CS.das_control["DAS_accState"]
       target_accel = actuators.accel
       target_speed = max(CS.out.vEgo + (target_accel * CarControllerParams.ACCEL_TO_SPEED_MULTIPLIER), 0)
       max_accel = 0 if target_accel < 0 else target_accel
       min_accel = 0 if target_accel > 0 else target_accel
 
-      while len(CS.das_control_counters) > 0:
-        can_sends.extend(self.tesla_can.create_longitudinal_commands(CS.acc_state, target_speed, min_accel, max_accel, CS.das_control_counters.popleft()))
+      counter = CS.das_control["DAS_controlCounter"]
+      can_sends.append(self.tesla_can.create_longitudinal_commands(acc_state, target_speed, min_accel, max_accel, counter))
 
     # Cancel on user steering override, since there is no steering torque blending
     if hands_on_fault:
       pcm_cancel_cmd = True
 
-    if self.frame % 10 == 0 and pcm_cancel_cmd:
-      # Spam every possible counter value, otherwise it might not be accepted
-      for counter in range(16):
-        can_sends.append(self.tesla_can.create_action_request(CS.msg_stw_actn_req, pcm_cancel_cmd, CANBUS.chassis, counter))
-        can_sends.append(self.tesla_can.create_action_request(CS.msg_stw_actn_req, pcm_cancel_cmd, CANBUS.autopilot_chassis, counter))
+    # Sent cancel request only if ACC is enabled
+    if self.frame % 10 == 0 and pcm_cancel_cmd and CS.acc_enabled:
+      counter = int(CS.sccm_right_stalk["SCCM_rightStalkCounter"] + 1) % 16
+      can_sends.append(self.tesla_can.right_stalk_press(counter, 1))  # half up (cancel acc)
+      can_sends.append(self.tesla_can.right_stalk_press((counter + 1) % 16, 0))  # to prevent neutral gear warning
```

Comment:
little weird that it's `% 16` here again, let's make the initial counter variable be the actual current counter, and then `+1` and `+2` with the `% 16` on each line